content/post/danielepo/pills/gestire-le-versioni-con-gitversion-introduzione.it.md

```diff
@@ -0,0 +1,102 @@
+
+Per ora ci concentriamo sull'installazione e sull'utilizzo in locale e in un prossimo video vedremo come utilizzarlo nella nostra pipeline di build e rilascio.
+
+Nel sito di documentazione di gitversion questo viene definito semplicemente come uno strumento ti aiuta ad implementare il semantic versioning nei tuoi progetti. Sebbene questa reazione riassuma perfettamente l'utilità di gitversion, per meglio apprezzarlo è meglio vederlo in azione. Procediamo quindi ad installarlo
+
+Come vi dicevo prima voglio usarlo localmente e successivamente vedremo altri metodi di utilizzo, quindi per trovare le istruzioni su come installarlo vado nella pagina "Getting started e poi Command line"
+
+Qua vedete che potete installare git version con Chocolatey se siete su windows oppure con homebrew se siete su Mac. Io sono su windows e quindi vi mostro come installarlo con chocolatey (per inciso, se non usate chocolatey non sapete cosa vi perdete)
+
+Semplicemente apro powershell in modalità amministratore e lancio il commando `choco install gitversion.portable -y` e premo invio.  Possiamo vedere che l'istallazione è partita ed una volta completata posso confermare che l'installazione sia andata a buon fine lanciando il comando  `gitversion /version`
+
+Un modo alternativo che non richiede l'utilizzo di un package manager, nel caso non siate abilitati ad usarlo è quello di scaricare il pacchetto nuget GitVersion.CommandLine e dentro trovate l'eseguibile. Questa procedura non è documentata da nessuna parte quindi prendetela con le pinzette
+
+Ora che l'abbiamo installato sulla nostra macchina è il momento di cominciare a trarne dei vantaggi. 
+
+Per vederlo in azione creiamo un progetto dotnet core con la CLI , niente di esoterico, una semplice console application. Facciamo `dotnet new console` e vediamo che viene creato il programma, poi con `dotnet run` lo eseguiamo e vediamo il classico "Hello world!". 
+
+Ora, per usare gitversion abbiamo bisogno di … git, quindi inizializziamo un repositori con `git init`. Andiamo su github punto com slash github shash gitignore e scarichiamo un gitignore adatto al nostro ambiente. Scegliamo visualstudio e lo salviamo direttamente nella cartella del progetto togliendo tutto quello che sta prima di puntogitignore.
+
+ Ora possiamo fare `git add .` e `git commit -m "first commit"`
+
+Ora che abbiamo un repository con almeno un commit proviamo ad eseguire gitversion. Possiamo da subito vedere che il commando torna una serie di informazioni sulla versione. Prima di tutto come default propone la versione zero punto uno punto zero. 
```

Comment:
qui, come dappertutto, sicuro di voler usare testo?? "zero punto uno punto zero" sembra uno speech to text :-)